### ISSUE TITLE
header-present check does not filter by true headings

### ISSUE THREAD

**Issue by straker** [dequelabs/axe-core#2146]
The header-present check looks for any `h1, h2, h3, h4, h5, h6, [role="heading"]` element but does not ensure that those heading elements are headings (and don't have an overriding role on them).

I.E. `<h1 role="none">`

**Comment by padmavemulapati** [comment]
Verified with axe-core3.5.5(latest code), and the test code is `<h1 role="none" id="target">Some stuff and stuff</h1>` , as per 
`https://github.com/dequelabs/axe-core/blob/695ec040d0269bb9a2e4125446d639d690975628/lib/rules/bypass.json#L18` 
` "any": ["internal-link-present", "header-present", "landmark"], ` this `header-present` check  should come with `bypass` ruleID.
But I am not seeing anything with `bypass` ruleID.
@straker , can you please check once.

**Comment by padmavemulapati** [comment]
@straker can you please check this? or QA notes please anything else required other than the above comment content?

**Comment by straker** [comment]
@padmavemulapati The bypass rule requires a link on the page, so you'll need to test with one

```html
<a href="foo">foo</a>
<h1 role="none" id="target">Some stuff and stuff</h1>
```

**Comment by padmavemulapati** [comment]
Verified with the given test file - seeing the bypass rule failure

![image](https://user-images.githubusercontent.com/47409579/90120089-02bd5d00-dd78-11ea-972b-ebcc0eaffd28.png)
